src/witan/send/driver/ingest/valid_states.clj

```diff
@@ -1,7 +1,30 @@
 (ns witan.send.driver.ingest.valid-states
   (:require [clojure.data.csv :as csv]
             [clojure.java.io :as io]
-            [clojure.string :as s]))
+            [clojure.string :as s]
+            [witan.send.driver.ingest.excel :as ie]
+            [witan.send.driver.ingest :as i]))
+
+(defn clip [n lower-bound upper-bound]
+  (cond
+    (< n lower-bound) lower-bound
+    (> n upper-bound) upper-bound
+    :else n))
+
+(def valid-states-xf
+  (comp (map #(update % :min-academic-year i/->int))
+        (map #(update % :max-academic-year i/->int))
+        (map #(assoc % :setting-group (:setting %)))
+        (map #(assoc % :max-academic-year (clip (:max-academic-year %) -5 21)))))
+
+(defn load-valid-states [file-name fixup-xf]
+  (let [header [:setting :setting-group :min-academic-year :max-academic-year :needs :setting->setting]]
+    (into []
+          (comp
+           (drop 1)
+           fixup-xf)
+          (ie/rows file-name "Sheet1" ;;(zipmap [:A :B :C :D :E :F] header)
```

Comment:
what is the commented out bit doing here?

Reply:
who knows...